src/hotspot/share/compiler/directivesParser.cpp

```diff
@@ -102,6 +102,7 @@ bool DirectivesParser::parse_from_file_inner(const char* filename, outputStream*
         ::close(file_handle);
         return parse_string(buffer, stream) > 0;
       }
+      ::close(file_handle);
```

Comment:
You could just move the existing `::close` to before the `if` statement ie close it immediately after the read.

Reply:
I agree and have pushed another commit with your suggestion.